packages/wonder-blocks-tooltip/components/tooltip-bubble.js

```diff
@@ -1,77 +1,69 @@
-    _renderBubble(childrenProps: PopperChildrenProps) {
-        const {children, placement} = this.props;
-        const {ref, outOfBoundaries, style, arrowProps} = childrenProps;
-        const actualPlacement = childrenProps.placement || placement;
+    _lastRef: ?HTMLElement;
 
-        // Here we take the props provided to us by popper.js and use them
-        // to position ourselves. We don't use a View because the ref callback
-        // for the react-popper integration expects an HTMLElement and coercing
-        // a View ref to the right type is a lot of extra lifting for little
-        // gain over just using a div.
-        return (
-            <div
-                data-placement={actualPlacement}
-                ref={ref}
-                className={css(styles.bubble, outOfBoundaries && styles.hide)}
-                style={style}
-            >
-                <View style={styles[`content-${actualPlacement}`]}>
-                    {children}
-                    <TooltipArrow
-                        placement={actualPlacement}
-                        popperArrowProps={arrowProps}
-                    />
-                </View>
-            </div>
-        );
+    _updateRef(ref: ?(React.Component<*> | Element)) {
+        const {popperProps} = this.props;
+        // We only want to update the popper's arrow reference if it is
+        // actually changed. Otherwise, we end up in an endless loop of updates
+        // as every render would trigger yet another render.
+        if (popperProps && ref) {
+            const domNode = ReactDOM.findDOMNode(ref);
+            if (domNode instanceof HTMLElement && domNode !== this._lastRef) {
+                this._lastRef = domNode;
+                popperProps.ref(domNode);
+            }
+        }
     }
 
     render() {
-        const {anchorElement, placement} = this.props;
+        const {children, popperProps} = this.props;
+        const {placement, outOfBoundaries, style, arrowProps} = popperProps;
+
         return (
-            <Popper
-                referenceElement={anchorElement}
-                placement={placement}
-                modifiers={{
-                    wbVisibility: visibilityModifierDefaultConfig,
-                    preventOverflow: {boundariesElement: "viewport"},
-                }}
+            <View
+                data-placement={placement}
+                ref={(r) => this._updateRef(r)}
+                style={[
+                    style,
+                    outOfBoundaries && styles.hide,
+                    styles.bubble,
+                    styles[`content-${placement}`],
+                ]}
             >
-                {(props) => this._renderBubble(props)}
-            </Popper>
+                {children}
+                <TooltipArrow
+                    placement={placement}
+                    popperArrowProps={arrowProps}
+                />
+            </View>
```

Comment:
So now were just rendering the bubble and the `popper` stuff is elsewhere.

Reply:
Yup